src/directives/nvd3Directives.js

```diff
@@ -1218,8 +1218,8 @@
                                 generate: function(){
                                     initializeMargin(scope, attrs);
                                     var chart = nv.models.pieChart()
-                                        .x(attrs.x === undefined ? function(d){ return d[0]; } : scope.x())
-                                        .y(attrs.y === undefined ? function(d){ return d[1]; } : scope.y())
+                                        .x(attrs.x === undefined ? function(d){ return d.key; } : scope.x())
```

Comment:
I believe this change will break existing users of the directives.

Instead, can't the user just provide their own x() and y() methods which can pull the data via their own defined attributes?  That is what we are doing.